### PR TITLE
Fix/mypy

### PR DESCRIPTION
**Pull request by tribunsky-kir**

### CODE REVIEW

File: blocks/db/next/sql.py [hunk 1]
```diff
@@ -42,50 +41,58 @@ class Dialects(str, Enum):
...
         logger.warning(' '.join(message))
-        self.data = {k: [] for k in KEYWORDS}
-        self._whereable = set()
-        self.codec: Type[Union[Row, Table]] = codec
-        self._fields = []
+        self.data: Dict[str, List[str]] = {k: [] for k in KEYWORDS}
+        self._where: Set[str] = set()
+        self._codec: Optional[Type[Union[Event, Table]]] = codec
+        self._fields: List[str] = []
         self._placeholder = _PLACEHOLDERS[dialect]
```
Comment (Sehat1137) [inline blocks/db/next/sql.py:55]: You set type for all properties except self._placeholder.
May be for it need add too, to save uniformity?

File: blocks/db/next/types.py [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            for row in self.rows:
+                # if is_dataclass(row):
+                #     dct = asdict(row)
+                #     tpl = tuple(dct[col] for col in fields)
+                #     values.append(tpl)
+                # elif isinstance(row, BaseModel):
+                #     dct = row.dict()
+                #     tpl = tuple(dct[col] for col in fields)
+                #     values.append(tpl)
+                # else:
```
Comment (Sehat1137) [inline blocks/db/next/types.py:58]: Just delete this or uncomment

File: blocks/db/next/types.py [hunk 3]
```diff
@@ -0,0 +1,71 @@
...
+                values.append(tpl)
+            return values
+        return []
+
+    @classmethod
+    def type(cls) -> Type[Event]:
+        # pydantic:
+        # row_codec = query.codec.__annotations__['rows'].__args__[0]
+        # dataclass:
+        # return cls.rows.__args__[0]
```
Comment (Sehat1137) [inline blocks/db/next/types.py:69]: Just delete this or uncomment

File: blocks/sources/parallel_processor.py [hunk 4]
```diff
@@ -1,6 +1,6 @@
-from blocks.types import ParallelEvent
+from blocks.types import EventOrEvents, ParallelEvent
 
 
-def run_parallel_processor(payload: bytes):
+def run_parallel_processor(payload: bytes) -> EventOrEvents:
```
Comment (Sehat1137) [inline blocks/sources/parallel_processor.py:4]: Optional[EventOrEvents]

File: tests/redis/test_smoke.py [hunk 5]
```diff
@@ -0,0 +1,85 @@
...
+    from redis import Redis
+except ImportError:
+    pytest.skip("skipping redis-only tests", allow_module_level=True)
+from pytest import approx
+from pydantic import BaseModel
+
+from blocks.redis import InputStream, RedisConsumer
+
+
+class RedisMock(Redis):
```
Comment (Sehat1137) [inline tests/redis/test_smoke.py:16]: It`s not real mock object. I think better name will be RedisStub

File: blocks/db/next/sql.py (hunk 1 above)
Comment (tribunsky-kir) [reply blocks/db/next/sql.py:55]: Those attributes were required mostly by mypy. 

I'll add `self._placeholder`'s type hint for consistency.

File: blocks/db/next/types.py (hunk 3 above)
Comment (tribunsky-kir) [reply blocks/db/next/types.py:56]: It's mostly comment for developers, cause there are different ways to extract actual type.

Not sure which is better: get type from instance in list (won't work with empty tables) or from type hint (easy to break w/o pydantic).

**Pull request merged**